jetty9/src/main/java/com/thoughtworks/go/server/GoServerLoadingIndicationHandler.java

```diff
@@ -0,0 +1,116 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.server;
+
+import com.thoughtworks.go.util.GoConstants;
+import com.thoughtworks.go.util.SystemEnvironment;
+import org.apache.commons.io.IOUtils;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.MimeTypes;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+import org.eclipse.jetty.server.handler.ContextHandler;
+import org.eclipse.jetty.webapp.WebAppContext;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import static com.thoughtworks.go.util.SystemEnvironment.LOADING_PAGE;
+import static org.eclipse.jetty.http.MimeTypes.Type.*;
+
+/** When GoCD is starting. This is the only handler that will be active (till the web application context handler is up).
+ *  During that time, this handler shows a 503 for all requests, while waiting for the rest of the server to be up.
+ */
+class GoServerLoadingIndicationHandler extends ContextHandler {
+    private WebAppContext webAppContext;
+    private boolean isWebAppStarting;
+    private SystemEnvironment systemEnvironment;
+
+    GoServerLoadingIndicationHandler(WebAppContext webAppContext, SystemEnvironment systemEnvironment) {
+        setContextPath("/");
+        setHandler(new LoadingHandler());
+
+        this.webAppContext = webAppContext;
+        this.isWebAppStarting = !webAppContext.isAvailable();
+        this.systemEnvironment = systemEnvironment;
+    }
+
+    private class LoadingHandler extends AbstractHandler {
+        @Override
+        public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+            if (isWebAppStarting()) {
+                handleQueriesWhenWebAppIsStarting(target, baseRequest, request, response);
+            } else if ("/".equals(target)) {
+                addHeaders(response);
+                response.sendRedirect(GoConstants.GO_URL_CONTEXT + systemEnvironment.landingPage());
+            }
+        }
+
+        private void handleQueriesWhenWebAppIsStarting(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException {
+            if (acceptHeaderValue(baseRequest).contains("json")) {
```

Comment:
This check is purposefully simple. Doesn't really do a great job of verifying that the accept header is really mentioning JSON, etc. I think this is fine. Tests reflect this.